driver/src/main/java/org/neo4j/driver/internal/async/NetworkSession.java

```diff
@@ -194,7 +196,7 @@ public boolean isOpen()
                 if ( cursor != null )
                 {
                     // there exists a cursor with potentially unconsumed error, try to extract and propagate it
-                    return cursor.failureAsync();
+                    return cursor.consumeAsync();
```

Comment:
This is one of the places I mentioned above… 
Wouldn't that change be applicable to other places where `failureAsync` is used?